docs/adr/0003-graphql-layer-between-backend-frontend.md

```diff
@@ -0,0 +1,98 @@
+# GraphQL layer between backend and frontend
+
+Date: 2020-04-30
+
+## Status
+
+**Accepted**
+
+## Related tickets (not available for public)
+
+SIG-2568
+
+## Context
+
+Since our current backend only serves data through REST and there is no possibility to generate
+types and or interfaces, there is no also no automated way to validate data on the frontend
+which is coming from the backend.
+
+Besides of that there are too many inconstencies between different backend calls and the current
+backend API documentation is not always correct and/or up to date because it's written manually
+and not generated from code.
+
+Because of this our `React components` are overcomplicated since we need to transform the API output to a
+format that the API can consume and that leads to bloat and too much client-side logic.
+
+## Proposal
+
+We would like to introduce a thin layer between the backend and our frontend which should expose
+backend data through GraphQL Queries and should expose backend PUT/POST/DELETE actions via GraphQL
+mutations.
+
```

Comment:
Agree with @bartjkdp also like I said before we need to identify which problems the FE is facing and how we can tackle these problems.

Also if we decide to implement GraphQL this must live in the API and not a "thin layer" that exists between the FE and API.

Reply:
Can you elaborate a bit more on why you think that the BFF should live on the back-end side of the architecture?

Reply:
@janjaap it lets us skip one transformation:
- DB -> business logic / REST -> BFF -> FE  (as proposed in this ADR)
vs
- DB -> business logic -> GraphQL -> FE
- DB -> business logic -> REST -> external partners

The business logic can than be shared between the REST interface we have to maintain and the new GraphQL interface.

Reply:
As a PS: 
- Implementing a GraphQL interface in the backend also avoids a network hop and removes the need for caching in the BFF layer.